src/py/mat3ra/made/tools/build/defect/configuration.py

```diff
@@ -54,20 +55,37 @@ def from_site_id(
         return cls(crystal=crystal, defect_type=defect_type, coordinate=coordinate, chemical_element=chemical_element)
 
     @classmethod
-    def from_approximate_position(
+    def from_approximate_coordinate(
         cls,
         crystal: Material,
         defect_type: PointDefectTypeEnum,
         approximate_coordinate: List[float],
         chemical_element: Optional[str] = None,
+        use_cartesian_coordinates: bool = False,
     ):
         if not crystal:
             raise RuntimeError("Crystal is not defined")
-        closest_site_id = get_closest_site_id_from_coordinate(crystal, approximate_coordinate)
+        closest_site_id = get_closest_site_id_from_coordinate(
+            material=crystal,
+            coordinate=approximate_coordinate,
+            use_cartesian_coordinates=use_cartesian_coordinates,
+        )
         return cls.from_site_id(
             crystal=crystal, defect_type=defect_type, site_id=closest_site_id, chemical_element=chemical_element
         )
 
+    @classmethod
+    def from_dict(cls, crystal: Material, data: dict):
+        if "site_id" in data:
+            config = PointDefectConfiguration.from_site_id(crystal=crystal, **data)
+        elif "coordinate" in data:
+            config = PointDefectConfiguration(crystal=crystal, **data)
+        elif "approximate_coordinate" in data:
+            config = PointDefectConfiguration.from_approximate_coordinate(crystal=crystal, **data)
+        else:
+            raise ValueError(f"Invalid defect configuration: {data}")
+        return config
```

Comment:
Constructor needs to return `cls()`. We can override the default constructor with it.